plugins/openai_plugin.py

```diff
@@ -177,6 +185,88 @@ def request_http(self, query: dict, user_id: int, test_end_time: float = 0):
+        
+        if len(queries) == 1:
+            UserWarning("Single prompt requests must not be run with the batch API")
+        
+        result = RequestResult(
+            user_id = user_id,
+            input_id = [query.get("input_id") for query in queries],
+            input_tokens = [query.get("input_tokens") for query in queries],
+            )
+
+        headers = {
+            "Content-type": "application/json"
+        }
+
+        if "chat/completions" in self.host:
+            raise ValueError("Batch requests are supported only in v1/completions API")
+        
+        data = {
+            "prompt" : [query.get("text") for query in queries],
+            # TODO: Evaluate the logical option to select the max and min token lengths. We are playing the safer option at the moment. 
+            "max_tokens": max([query.get("output_tokens") for query in queries]), 
+            "min_tokens": min([query.get("output_tokens") for query in queries]),
+        }
+        data.update(self.request_defaults)
+
+        if self.model_name is not None:
+            data["model"] = self.model_name
+        
+        response = None
+        try:
+            result.start_time = time.time()
+            response = requests.post(self.host, headers=headers, json=data, verify=False)
+            response.raise_for_status()
+            result.end_time = time.time()
+        except requests.exceptions.ConnectionError as err:
+            result.end_time = time.time()
+            result.error_text = repr(err)
+            if response is not None:
+                result.error_code = response.status_code
+            logger.exception("Connection Error")
+            return result
+        except requests.exceptions.HTTPError as err:
+            result.end_time = time.time()
+            result.error_text = repr(err)
+            if response is not None:
+                result.error_code = response.status_code
+            logger.exception("HTTP Error")
+            return result
+
+        logger.debug("Response: %s", json.dumps(response.text))
+
+        try:
+            message = json.loads(response.text)
+            error = message.get("error")
+
+            if error is None:
+                if len(message["choices"]) != len(result.input_id):
+                    raise IndexError("Returned number of sequences does not match the sequences sent")
+                
+                result.output_tokens = message["usage"]["completion_tokens"]
+                result.input_tokens = message["usage"]["prompt_tokens"]
+                # TODO: Raise issue on incorrect naming - OpenAI API returns both a finish reason and a stop reason. 
+                # stop_reason indicates a possible failure of the sequence. finish_reason could also indicate that we reached the requested length.
+                result.stop_reason = [choice.get('finish_reason') for choice in message["choices"]] 
```

Comment:
Likewise to above, `"choices"` may be a KeyError.

Reply:
Resolved in [Fix : Potential KeyErrors](https://github.com/openshift-psap/llm-load-test/pull/66/commits/d32611035fdba1e265c270a76a8c1e59da69e7e7)